get-started/integrations/llama-ocr.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Integrate Llama OCR with kluster.ai API
+description: Learn how to use kluster.ai as an LLM provider for the Llama OCR NPM library to run vision-enabled OCR with any multimodal model on the kluster.ai platform.
+---
+
+# Integrate Llama OCR with kluster.ai
+
+[Llama OCR](https://llamaocr.com/){target=\_blank} is an NPM library that converts images into richly formatted Markdown by sending the image and a conversion prompt to a vision-capable LLM. Llama OCR is compatible with the [kluster.ai](https://www.kluster.ai/){target=\_blank} API with just a few changes. Note that not all models supported by kluster.ai are capable of processing images. The [vision-capable models](/get-started/models/){target=\_blank} currently supported by kluster.ai include:
+
+- `google/gemma-3-27b-it` - Google's Gemma 3 model with vision capabilities
+- `Qwen/Qwen2.5-VL-7B-Instruct` - Qwen's visual language model
+- `meta-llama/Llama-4-Maverick-17B-128E-Instruct-FP8` - Meta's Llama 4 Maverick model
+- `meta-llama/Llama-4-Scout-17B-16E-Instruct` - Meta's Llama 4 Scout model
+
+This guide walks you through the minimal changes needed to:
+
+- Clone the repo and install the dependencies
+- Configure kluster.ai as the provider
+- Run a working end-to-end OCR example against the sample grocery receipt image that ships with the repo
+
+## Prerequisites
+
+--8<-- 'text/kluster-api-onboarding.md'
+- Node version 18 or higher
+
+You can clone the Llama OCR repo and installed the required dependencies with the following commands:
+
+```bash
+git clone https://github.com/Nutlope/llama-ocr.git
+cd llama-ocr
+npm install
+```
+
+Additionally, you'll need to install the OpenAI package, which we'll use to interact with Kluster.ai's OpenAI-compatible API:
+
+```bash
+npm install openai
+```
+
+## Modify the OCR function
+
+Navigate to the `src` folder and find the `index.ts` file. Just a few changes need to be made here.
+
+1. Update the imports: 
+```typescript
+import OpenAI from "openai"; 
+import fs from "fs";                   
+```
+2. Modify the OCR function signature as follows: 
+```typescript
+export async function ocr({
+ filePath,
+ apiKey = "INSERT_API_KEY",         
+ model  = "google/gemma-3-27b-it",           
+}: {
+ filePath: string;
+ apiKey?: string;
+ model?: string;                             
+}) {
+```
+3. **Initialize the OpenAI client with kluster.ai endpoint**
+```typescript
+const openai = new OpenAI({
+ apiKey,
+ baseURL: "https://api.kluster.ai/v1",         
+});
+
+return getMarkdown({ openai, model, filePath });
+}
+```
+4. **Revise the getMarkdown function to use the OpenAI SDK** - The contents of the prompt remain the same but the function switches from the Together SDK to the OpenAI SDK, so it renames the client/model args accordingly and encodes local images inline with `fs.readFileSync` as follows:
```

Comment:
This is using different formatting from steps 1-3. So it should all be the same with `**Title** - description` or just description

Reply:
`getMarkdown`